src/main/java/org/variantsync/diffdetective/diff/difftree/DiffNode.java

```diff
@@ -110,7 +110,7 @@ public DiffNode(DiffType diffType, NodeType nodeType,
     public static DiffNode createRoot() {
         return new DiffNode(
                 DiffType.NON,
-                NodeType.ROOT,
+                NodeType.IF,
```

Comment:
This change causes duplicate IDs on the following diff:
```diff
 #ifdef A
  Code
  #endif
  ```
The root node and its only child have the same ID, because both have the following values:
- from line number: `DiffLineNumber(1, 1, 1)`
- NodeType `NodeType.IF`
- DiffType: `DiffType.NON`

Suggestion by @pmbittner : Change the line number of the root node to `DiffLineNumber.Invalid`.

Reply:
Good catch! We could change the line number `from` to `DiffLineNumber.Invalid` but we have to make sure that this does not cause trouble with the bit arithmetics. I remember that I did not choose `DiffLineNumber.Invalid` for `from` of the root back then but I don't remember the reason. Maybe it was because of the bit arithmetics or something else. Another option could be to keep the DiffType `ROOT` but only use it for ID conversion and disallow its use as a node's nodeType. Then in `getID`, a node could use ROOT if it is the root and its nodeType otherwise.

Reply:
A negative line number would be less hacky though and would also highlight that a root that was created by `createRoot` is synthetic and not an actual part of the diffed artifact.

Reply:
The `DiffNode.getID` does add 1 to the line number so negative/invalid line numbers don't cause issues in computing the IDs. Furthermore, he `inDiff` line numbers should be unique for all nodes (no two nodes can be parsed from a single line). Thus, all `DiffNode.getID` should be unique again.

Reply:
Ok! So `inDiff` should indeed be `DiffLineNumber.Invalid()`. Would you write a unit test to assert that the IDs of an example DiffTree are always unique?

Reply:
A test for this follows in the parser refactoring pull request.